pymc/distributions/continuous.py

```diff
@@ -568,13 +568,13 @@ def rng_fn(
         size: Optional[Union[List[int], int]],
     ) -> np.ndarray:
         return stats.truncnorm.rvs(
-            a=(lower - mu) / sigma,
-            b=(upper - mu) / sigma,
-            loc=mu,
-            scale=sigma,
+            a=((lower - mu) / sigma).astype("float64"),
+            b=((upper - mu) / sigma).astype("float64"),
+            loc=(mu).astype("float64"),
+            scale=(sigma).astype("float64"),
             size=size,
             random_state=rng,
-        )
+        ).astype(mu.dtype)
```

Comment:
```suggestion
        ).astype(self.dtype)
```

Reply:
Seems like `rng_fn` is a classmethod. I'm not sure if this would work. But I admit 
`mu.dtype` is a bit ugly

Reply:
Implemented the mapped dtype in https://github.com/pymc-devs/pymc/pull/7026/commits/82228a903fc11965698ba8c62192b5bbed0ceba8